spec/support/helpers.rb

```diff
@@ -4,6 +4,7 @@ module Helpers
   def with_isolated_config
     original_config = RSpec.configuration
     RSpec.configuration = RSpec::Core::Configuration.new
+    RSpec.configuration.world = RSpec.world
```

Comment:
The whole point of isolated configuration is to be isolated from the global test run, we don't want the child configuration sharing the parent world, else those specs get counted in various places. What were you seeing that made you want to do this?

Reply:
Sorry, I disagree. `with_isolated_config` sounds like the configuration is isolated, not that what is inside the block should be in a different world.

[We wrap each example](https://github.com/rspec/rspec-rails/blob/71c43641759b41ec26e7e8b85d285c20f1eabb14/spec/spec_helper.rb#L54) to run in its own world:
```ruby

  config.around(:example) do |example|
    real_world = RSpec.world
    RSpec.instance_variable_set(:@world, RSpec::Core::World.new)
    example.run
    RSpec.instance_variable_set(:@world, real_world)
  end
```

Also, all current usages of `with_isolated_config` wrap the whole example it is used in, there are no separate "inside" and "outside" example parts that would be prone to affect each other.

WDYT of making `with_isolated_config` a metadata-induced `around` hook to make this more obvious?

Reply:
Wow I had no idea we did that, ok, well you need that world, I don't think you can rely on the global here...

Reply:
So it seems that the real problem, that prevents us from using a unique isolated configuration is [in `RSpec::Core::ExampleGroup.describe`](https://github.com/rspec/rspec-core/blob/a01aba79715dda8ede1bdb7d1c5611094939fc1d/lib/rspec/core/example_group.rb#L260):
```ruby
RSpec.world.example_groups
```

Should it be a `RSpec.configuration.world.example_groups`?

This results in:

![image](https://user-images.githubusercontent.com/6916/88476512-994cea00-cf41-11ea-84e5-0b5c77601b37.png)

> I don't think you can rely on the global here...

The whole point is that I don't rely on the global, real one. The global is kept in `real_world` during the examples, and `RSpec.world` is assigned (using `instance_variable_set`) a `World.new`.
But `RSpec::Core::Configuration.new` gets its own world:
```ruby
  def with_isolated_config
    original_config = RSpec.configuration
    RSpec.configuration = RSpec::Core::Configuration.new
```
which is an `RSpec::Core::World::Null`.

I strongly doubt it would make sense to change the Core to:
```diff
- RSpec.world.example_groups
+ RSpec.configuration.world.example_groups
```

But in this case, setting an isolated world initialized specifically for this example to an isolated config that was also initialized for this example makes total sense.

Reply:
Can you drop this change for now, and open another PR to work on this? I'm happy to merge the rest of these changes, but I'd like to look at this a bit more

Reply:
Makes sense. Done.